core/src/main/java/com/graphhopper/routing/Router.java

```diff
@@ -159,7 +160,7 @@ private void checkIfPointsAreInBounds(List<GHPoint> points) {
     }
 
     private void checkHeadings(GHRequest request) {
-        if (request.getHeadings().size() > 1 && request.getHeadings().size() != request.getPoints().size())
```

Comment:
was this intended or a bug?

Reply:
You mean it should be `>0`? I think we want to check that there are either no headings or as many as there are points.

Reply:
Ah no this is correct. We can specify a start heading also for routing with via points (without specifying a heading for all via points).